lib/Models/CzmlCatalogItem.ts

```diff
@@ -1,20 +1,75 @@
 import i18next from "i18next";
-import { computed, toJS } from "mobx";
+import { action, computed, observable, toJS } from "mobx";
+import Clock from "terriajs-cesium/Source/Core/Clock";
+import JulianDate from "terriajs-cesium/Source/Core/JulianDate";
 import CzmlDataSource from "terriajs-cesium/Source/DataSources/CzmlDataSource";
 import isDefined from "../Core/isDefined";
 import { JsonObject } from "../Core/Json";
 import readJson from "../Core/readJson";
 import TerriaError from "../Core/TerriaError";
 import AsyncMappableMixin from "../ModelMixins/AsyncMappableMixin";
 import CatalogMemberMixin from "../ModelMixins/CatalogMemberMixin";
+import DiscretelyTimeVaryingMixin from "../ModelMixins/DiscretelyTimeVaryingMixin";
 import UrlMixin from "../ModelMixins/UrlMixin";
 import CzmlCatalogItemTraits from "../Traits/CzmlCatalogItemTraits";
 import CreateModel from "./CreateModel";
+import LoadableStratum from "./LoadableStratum";
 import Mappable from "./Mappable";
+import { BaseModel } from "./Model";
+import proxyCatalogItemUrl from "./proxyCatalogItemUrl";
+import StratumOrder from "./StratumOrder";
+
+/**
+ * A loadable stratum for CzmlCatalogItemTraits that derives TimeVaryingTraits
+ * from the CzmlDataSource
+ */
+class CzmlTimeVaryingStratum extends LoadableStratum(CzmlCatalogItemTraits) {
+  static stratumName = "czmlLoadableStratum";
+
+  constructor(readonly catalogItem: CzmlCatalogItem) {
+    super();
+  }
+
+  duplicateLoadableStratum(model: BaseModel): this {
+    return new CzmlTimeVaryingStratum(model as CzmlCatalogItem) as this;
+  }
+
+  @computed
+  private get clock(): Clock | undefined {
+    return (this.catalogItem as any)._dataSource?.clock;
+  }
+
+  @computed
+  get currentTime(): string | undefined {
+    const currentTime = this.clock?.currentTime;
+    return currentTime ? JulianDate.toIso8601(currentTime) : undefined;
+  }
+
+  @computed
+  get startTime(): string | undefined {
+    const startTime = this.clock?.startTime;
+    return startTime ? JulianDate.toIso8601(startTime) : undefined;
+  }
+
+  @computed
+  get stopTime() {
+    const stopTime = this.clock?.stopTime;
+    return stopTime ? JulianDate.toIso8601(stopTime) : undefined;
+  }
+
+  @computed
+  get multiplier(): number | undefined {
+    return this.clock?.multiplier;
+  }
```

Comment:
If I understood it right, In v7, I every catalog item can have its own clock and the Terria timeline will use the clock of one of the items in the workbench. CZML used to expose the clock from `CzmlDataSource`:

https://github.com/TerriaJS/terriajs/blob/281c5e5c731264b8ebce7df48f5f6c65fbc6f250/lib/Models/CzmlCatalogItem.js#L189-L192

Whereas in v8, there is a global timeline clock and we set the start, end & current time from a catalog item that sits at the top of the timeline stack:
https://github.com/TerriaJS/terriajs/blob/ab494852613b198bd8418283d1064b05907c3bfc/lib/Models/TimelineStack.ts#L41-L56

This is exactly what is implemented in this PR. In the section you highlighted we ONLY derive the start, stop & current time from the clock in the CZML data source. We do this because we don't want to duplicate the CZML parsing that Cesium does.